packages/calling-stateful-client/src/CallAgentDeclarative.ts

```diff
@@ -1,26 +1,51 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { Call, CallAgent, CallEndReason, CollectionUpdatedEvent, IncomingCall } from '@azure/communication-calling';
+import {
+  Call,
+  CallAgent,
+  CallEndReason,
+  CollectionUpdatedEvent,
+  IncomingCallEvent
+} from '@azure/communication-calling';
 import { CallContext } from './CallContext';
 import { callDeclaratify, DeclarativeCall } from './CallDeclarative';
 import { CallSubscriber } from './CallSubscriber';
 import { convertSdkCallToDeclarativeCall, convertSdkIncomingCallToDeclarativeIncomingCall } from './Converter';
+import { DeclarativeIncomingCall, incomingCallDeclaratify } from './IncomingCallDeclarative';
 import { IncomingCallSubscriber } from './IncomingCallSubscriber';
 import { InternalCallContext } from './InternalCallContext';
-import { disposeAllViewsFromCall, disposeAllViews } from './StreamUtils';
+import { disposeAllViews, disposeAllViewsFromCall } from './StreamUtils';
+
+/**
+ * @public
+ * `DeclarativeCallAgent` extends and proxies the {@link @azure/communication-calling#CallAgent}
+ */
+export type DeclarativeCallAgent = CallAgent & {
+  /**
+   * @beta
+   * Used to return a declarative incoming call object for an incoming call id.
+   *
+   * @Remark This function doesn't exist on the {@link @azure/communication-calling#CallAgent} interface.
+   * @param incomingCallId - Incoming Call ID
+   * @returns proxied IncomingCall
+   */
+  /* @conditional-compile-remove(1-n-calling) */
```

Comment:
nitty nit: Move the conditional compilation directive above the docstring so that `tsdoc` isn't confused.